internal/graph/check.go

```diff
@@ -145,25 +143,14 @@ func WithMaxConcurrentReads(limit uint32) LocalCheckerOption {
 	}
 }
 
-func WithCachedResolver(opts ...CachedCheckResolverOpt) LocalCheckerOption {
-	return func(d *LocalChecker) {
-		cachedCheckResolver := NewCachedCheckResolver(d, opts...)
-		d.delegate = cachedCheckResolver
-	}
-}
-
-func WithSingleflightResolver() LocalCheckerOption {
-	return func(d *LocalChecker) {
-		singleflightResolver := NewSingleflightCheckResolver(d)
-		d.delegate = singleflightResolver
-	}
-}
-
 // NewLocalChecker constructs a LocalChecker that can be used to evaluate a Check
 // request locally.
-func NewLocalChecker(ds storage.RelationshipTupleReader, opts ...LocalCheckerOption) CheckResolver {
+//
+// The constructed LocalChecker is not wrapped with cycle detection. Developers
+// wanting a LocalChecker without other wrapped layers (e.g caching and others)
+// are encouraged to use [[NewLocalCheckerWithCycleDetection]] instead.
+func NewLocalChecker(opts ...LocalCheckerOption) *LocalChecker {
```

Comment:
This is confusing

```
wanting a LocalChecker without other wrapped layers (e.g caching and others)
// are encouraged to use [[NewLocalCheckerWithCycleDetection]] instead.
```

but NewLocalCheckerWithCycleDetection doesn't include support for caching. Did you mean to say `NewLayeredCheckResolver`?